### PR TITLE
Update REI + YACL compat; reintroduce Crafting Tweaks support (1.20)

### PR DESCRIPTION
**Pull request by unilock**
- Gradle, Fabric Loom, and all dependencies have been updated to their latest versions
  - ~~YACL is no longer `include`d in the JAR, as its documentation [strongly recommends against doing so](https://docs.isxander.dev/yet-another-config-lib/setup-with-gradle)~~
- REI support has been updated for REI 8.3+ (plugin is now entirely client-side)
- The config has been updated in accordance with YACL 3.2.0+, including usage of its config screen auto-generation feature
  - Translation keys have been updated accordingly
- Crafting Tweaks support has been reintroduced
- The [Fabric Transfer API](https://fabricmc.net/wiki/tutorial:transfer-api) has been implemented in `ProjectTableBlockEntity`, allowing the Project Table's inventory to be accessed via Hoppers and such
- The mod icons have been resized and optimized (via [`oxipng`](https://github.com/shssoichiro/oxipng)), decreasing their file size
- ~~The mod version now complies with [SemVer](https://semver.org)~~

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-# check these on https://modmuss50.me/fabric.html
+# Check these on https://fabricmc.net/develop/
 minecraft_version=1.20.1
-yarn_mappings=1.20.1+build.2
-loader_version=0.14.21
+yarn_mappings=1.20.1+build.10
+loader_version=0.14.25
+
 # Mod Properties
-mod_version=1.0+1.20.1
```
Comment (Luligabi1) [inline gradle.properties:10]: Please revert this; it's very useful for archiving multiple versions and it's fully supported by Fabric's mod schema. It's even used by [Fabric API](https://github.com/FabricMC/fabric/)

File: build.gradle [hunk 2]
```diff
@@ -32,53 +47,60 @@ dependencies {
 
     modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 
-    include modImplementation("dev.isxander.yacl:yet-another-config-lib-fabric:${project.yacl_version}") {
```
Comment (Luligabi1) [inline build.gradle:35]: Please revert this; I'm aware it's not recommended by the author, but my time modding and giving user support tells me it's in fact not likely the user will already have YACL on their mod folder.

**Review by Luligabi1 — CHANGES_REQUESTED** [review]
Looks great, just minor preferences I'd like reverted 😉

File: gradle.properties (hunk 1 above)
Comment (unilock) [reply gradle.properties:10]: The Minecraft version is automatically appended in `build.gradle`, making the end result practically identical to that of specifying the Minecraft version in the `mod_version` in `gradle.properties` manually, but with one fewer extra step.

https://github.com/Luligabi1/EnhancedWorkbenches/blob/7c9f07989413e29a894fc2f77bf60a614294c800/build.gradle#L6

I can still revert this change though, if you'd like.

Comment (Luligabi1) [reply gradle.properties:10]: ahh sorry, didn't see that. It *is* superior, but I'd probably forget this one mod has it done this way, ending up with stuff like `1.0+1.21+1.21` 😅 it's better to revert it

Comment (unilock) [reply gradle.properties:10]: Fair enough! I'll revert it then.

**Review by Luligabi1 — APPROVED** [review]
Thanks, I'll look into it tomorrow

**Pull request merged**